cli/command/stack/deploy.go

```diff
@@ -42,5 +42,7 @@ func newDeployCommand(dockerCli command.Cli) *cobra.Command {
 	flags.StringVar(&opts.ResolveImage, "resolve-image", swarm.ResolveImageAlways,
 		`Query the registry to resolve image digest and supported platforms ("`+swarm.ResolveImageAlways+`", "`+swarm.ResolveImageChanged+`", "`+swarm.ResolveImageNever+`")`)
 	flags.SetAnnotation("resolve-image", "version", []string{"1.30"})
+	flags.BoolVarP(&opts.Detach, "detach", "d", true, "Exit immediately instead of waiting for the stack services to converge")
+	flags.BoolVarP(&opts.Quiet, "quiet", "q", false, "Suppress progress output")
```

Comment:
It looks like this PR (currently at least) keeps the old behavior (`docker stack deploy` being non-interactive / detached as default).

If the plan is to eventually make "interactive" (`--detach=false`) the default, I also went looking how we handled the transition for `docker service create`; https://github.com/moby/moby/commit/21ec12b9676f95aaa69c3082b97d0b10031cc90a (https://github.com/moby/moby/pull/31144)

For that we chose to print a warning; wondering if (and if: when? / which release?) we want to do the same for `docker stack`; inform users that we plan to switch non-detach as default, with the option to return to the old (`--detach[=true]`) behavior.

Reply:
I've added a message just like in [moby/moby@21ec12b](https://github.com/moby/moby/commit/21ec12b9676f95aaa69c3082b97d0b10031cc90a) ([moby/moby#31144](https://github.com/moby/moby/pull/31144)) as you said, indicating that we plan to switch non-detach as default.